File: frontend/src/types/truck.ts
export interface FoodTruck {
  location_id: string | null;
  applicant: string | null;
  facility_type: string | null;
  location_description: string | null;
  address: string | null;
  status: string | null;
  food_items: string | null;
  latitude: number | null;
  longitude: number | null;
  days_hours: string | null;
}

export interface NearestTruckResult extends FoodTruck {
  distance_meters: number | null;
  distance_method: "haversine" | "google_maps";
}

File: backend/app/models/truck.py
from pydantic import BaseModel
from typing import Optional


class FoodTruck(BaseModel):
    location_id: Optional[str] = None
    applicant: Optional[str] = None
    facility_type: Optional[str] = None
    location_description: Optional[str] = None
    address: Optional[str] = None
    status: Optional[str] = None
    food_items: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    days_hours: Optional[str] = None


class NearestTruckResult(FoodTruck):
    distance_meters: Optional[float] = None
    distance_method: str = "haversine"  # "haversine" or "google_maps"